### PR TITLE
Fix CASA route popups

### PR DESCRIPTION
**Pull request by danji90**
- Converting individual LineStrings in RoutesLayer to one MultiLineString on fetch
- Prevents multiple popup rendering
- Added functionality in Map component for showPopupOnHover attribute to filter the features inside the individual featureInfos as a function

### CODE REVIEW

File: src/components/Map/Map.js [hunk 1]
```diff
@@ -128,10 +128,21 @@ class Map extends PureComponent {
...
-        infos = infos.filter(
-          ({ layer }) =>
-            layer.get('showPopupOnHover') && layer.get('popupComponent'),
-        );
+        infos = infos
+          .filter(
+            ({ layer }) =>
+              layer.get('showPopupOnHover') && layer.get('popupComponent'),
+          )
+          .map(i => {
```
Comment (oterral) [inline src/components/Map/Map.js]: call it `info`  we don' t understand what is `i`

File: src/components/Map/Map.js [hunk 2]
```diff
@@ -128,10 +128,21 @@ class Map extends PureComponent {
...
+            /* Apply showPopupOnHover function if defined to further filter features */
+            const showPopupOnHover = i.layer.get('showPopupOnHover');
+            if (typeof showPopupOnHover === 'function') {
+              const newFeature = i;
+              newFeature.features = i.layer.get('showPopupOnHover')(i.features);
+              return newFeature;
```
Comment (oterral) [inline src/components/Map/Map.js]:
```
return {
   ...info,
   features: i.layer.get('showPopupOnHover')(i.features)
}
```

File: src/layers/RouteLayer/RouteLayer.js [hunk 3]
```diff
@@ -26,7 +28,12 @@ class RouteLayer extends CasaLayer {
       }),
       ...options,
     });
-    this.set('showPopupOnHover', true);
+    this.set('showPopupOnHover', features => {
+      if (features.length) {
```
Comment (oterral) [inline src/layers/RouteLayer/RouteLayer.js]: You don't need the `if` you can apply the filter on an empty array

File: src/layers/RouteLayer/RouteLayer.js [hunk 4]
```diff
@@ -26,7 +28,12 @@ class RouteLayer extends CasaLayer {
       }),
       ...options,
     });
-    this.set('showPopupOnHover', true);
+    this.set('showPopupOnHover', features => {
```
Comment (oterral) [inline src/layers/RouteLayer/RouteLayer.js]:
```suggestion
    this.set('showPopupOnHover', (features = []) => {
```
to make sure it's always an array

File: src/components/Map/Map.js (hunk 1 above)
Comment (danji90) [reply src/components/Map/Map.js]: Done

**Pull request merged**